src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -224,27 +228,35 @@ internal abstract class AssetCache<TAssetEntry, TAsset> : AssetCache, IAssetCach
 
         public TAsset GetAsset(AssetId assetId)
         {
-            TAssetEntry assetEntry;
-            if (LookupByAssetId.TryGetValue(assetId, out assetEntry))
-            {
-                return assetEntry.Asset;
-            }
-            else
+            using (StateSynchronizationPerformanceMonitor.Instance.IncrementEventDuration(this.GetType().Name, "GetAsset"))
+            using (StateSynchronizationPerformanceMonitor.Instance.MeasureEventMemoryUsage(this.GetType().Name, "GetAsset"))

```

Comment:
>using (StateSynchronizationPerformanceMonitor.Instance.MeasureEventMemoryUsage(this.GetType().Name, "GetAsset")) [](start = 12, length = 112)

It would be surprising to me if this block or the GetAssetId block cause significant memory allocations. Is there some large allocation hidden here I'm not seeing?

Reply:
I can remove these for now, I was using these events for comparison when defer loading assets.